src/mount.ts

```diff
@@ -413,16 +413,33 @@ export function mount(
   // users expect stubs to work with globally registered
   // components, too, such as <router-link> and <router-view>
   // so we register those globally.
-  // https://github.com/vuejs/vue-test-utils-next/issues/249
+  // ref: https://github.com/vuejs/vue-test-utils-next/issues/249
+  // we register both the hyphenated version and non-hyphenated
+  // version for good measure.
+  // ref: https://github.com/vuejs/vue-test-utils-next/issues/425
   if (global?.stubs) {
     for (const [name, stub] of Object.entries(global.stubs)) {
       const tag = hyphenate(name)
       if (stub === true) {
+        const stubbed = createStub({ name, props: {} })
         // default stub.
-        app.component(tag, createStub({ name, props: {} }))
+        app.component(tag, stubbed)
+
+        if (
+          !name.includes('-') &&
+          !['transition', 'transition-group'].includes(name)
+        ) {
+          app.component(name, stubbed)
+        }
       } else {
         // user has provided a custom implementation.
         app.component(tag, stub)
+        if (
+          !name.includes('-') &&
+          !['transition', 'transition-group'].includes(name)
+        ) {
+          app.component(name, stub)
+        }
```

Comment:
Can't we simply register the stub with the provided key in stubs? That's what we do for components (and I realize that's why I used this in my apps instead of stubs) and it should work fine?

Something like (untested):
```ts
 for (const [name, stub] of Object.entries(global.stubs)) {
  if (stub === true) {
    // default stub.
    app.component(name, createStub({ name, props: {} }))
  } else {
    // user has provided a custom implementation.
    app.component(name, stub)
  }
```

Reply:
That is what I thought, but it was not working - seems there is something else at work here 🤔

Reply:
See comment below - I think we can just pascal case everything and it will "just work"